mypy/checker.py

```diff
@@ -1575,6 +1575,9 @@ def check_member_assignment(self, instance_type: Type, attribute_type: Type,
 
         Return the inferred rvalue_type and whether to infer anything about the attribute type
         """
+        if isinstance(instance_type, Instance) and isinstance(attribute_type, ClassVarType):
+            self.msg.fail("Illegal assignment to class variable", context)
+
```

Comment:
I believe this code should rather appear in ``analyze_member_access`` (there is also a special flag ``is_lvalue`` to indicate that an assignment to a member is being analysed rather that reading from it).